frontend/packages/knative-plugin/src/actions/providers.ts

```diff
@@ -91,6 +91,12 @@ export const useSinkPubSubActionProvider = (resource: K8sResourceKind) => {
 export const useKnativeServiceActionsProvider = (resource: K8sResourceKind) => {
   const [kindObj, inFlight] = useK8sModel(referenceFor(resource));
   const serviceTypeValue = React.useContext(KnativeServiceTypeContext);
+  const { loaded: podDataLoaded, podData } = usePodsWatcher(
+    resource,
+    referenceFor(resource),
+    resource.metadata.namespace,
+  );
+
```

Comment:
We have the resource already and it contains `status.conditions` we can use that to check function is ready or not. No need to fetch pods for that. Same `status.conditions` is used in Functions list page to show the status column. 

https://github.com/user-attachments/assets/370bdabd-f544-48f1-ace6-f549323c17fa